diff --git a/source/i18n/number_output.cpp b/source/i18n/number_output.cpp
index 1e86f2f9..6f4e2482 100644
--- a/source/i18n/number_output.cpp
+++ b/source/i18n/number_output.cpp
@@ -16,6 +16,7 @@ namespace number {
 
 UPRV_FORMATTED_VALUE_SUBCLASS_AUTO_IMPL(FormattedNumber)
 
+#define UPRV_NOARG
 
 UBool FormattedNumber::nextFieldPosition(FieldPosition& fieldPosition, UErrorCode& status) const {
     UPRV_FORMATTED_VALUE_METHOD_GUARD(FALSE)
@@ -30,12 +31,12 @@ void FormattedNumber::getAllFieldPositions(FieldPositionIterator& iterator, UErr
 
 void FormattedNumber::getAllFieldPositionsImpl(FieldPositionIteratorHandler& fpih,
                                                UErrorCode& status) const {
-    UPRV_FORMATTED_VALUE_METHOD_GUARD()
+    UPRV_FORMATTED_VALUE_METHOD_GUARD(UPRV_NOARG)
     fData->getStringRef().getAllFieldPositions(fpih, status);
 }
 
 void FormattedNumber::getDecimalQuantity(impl::DecimalQuantity& output, UErrorCode& status) const {
-    UPRV_FORMATTED_VALUE_METHOD_GUARD()
+    UPRV_FORMATTED_VALUE_METHOD_GUARD(UPRV_NOARG)
     output = fData->quantity;
 }
 
diff --git a/source/i18n/numrange_fluent.cpp b/source/i18n/numrange_fluent.cpp
index 079d5194..b284561c 100644
--- a/source/i18n/numrange_fluent.cpp
+++ b/source/i18n/numrange_fluent.cpp
@@ -377,6 +377,7 @@ LocalizedNumberRangeFormatter::getFormatter(UErrorCode& status) const {
 
 UPRV_FORMATTED_VALUE_SUBCLASS_AUTO_IMPL(FormattedNumberRange)
 
+#define UPRV_NOARG
 
 UBool FormattedNumberRange::nextFieldPosition(FieldPosition& fieldPosition, UErrorCode& status) const {
     UPRV_FORMATTED_VALUE_METHOD_GUARD(FALSE)
@@ -391,7 +392,7 @@ void FormattedNumberRange::getAllFieldPositions(FieldPositionIterator& iterator,
 
 void FormattedNumberRange::getAllFieldPositionsImpl(
         FieldPositionIteratorHandler& fpih, UErrorCode& status) const {
-    UPRV_FORMATTED_VALUE_METHOD_GUARD()
+    UPRV_FORMATTED_VALUE_METHOD_GUARD(UPRV_NOARG)
     fData->getStringRef().getAllFieldPositions(fpih, status);
 }
 
