diff --git a/source/i18n/number_decimfmtprops.h b/source/i18n/number_decimfmtprops.h
index a179d087..3e25966b 100644
--- a/source/i18n/number_decimfmtprops.h
+++ b/source/i18n/number_decimfmtprops.h
@@ -16,8 +16,6 @@
 #include "number_types.h"
 
 U_NAMESPACE_BEGIN
-namespace number {
-namespace impl {
 
 // Export an explicit template instantiation of the LocalPointer that is used as a
 // data member of CurrencyPluralInfoWrapper.
@@ -29,6 +27,9 @@ template class U_I18N_API LocalPointerBase<CurrencyPluralInfo>;
 template class U_I18N_API LocalPointer<CurrencyPluralInfo>;
 #endif
 
+namespace number {
+namespace impl {
+
 // TODO: Figure out a nicer way to deal with CurrencyPluralInfo.
 // Exported as U_I18N_API because it is a public member field of exported DecimalFormatProperties
 struct U_I18N_API CurrencyPluralInfoWrapper {
diff --git a/source/i18n/number_patternmodifier.h b/source/i18n/number_patternmodifier.h
index 3679ad1f..705037f0 100644
--- a/source/i18n/number_patternmodifier.h
+++ b/source/i18n/number_patternmodifier.h
@@ -15,11 +15,6 @@
 #include "number_utils.h"
 
 U_NAMESPACE_BEGIN
-namespace number {
-namespace impl {
-
-// Forward declaration
-class MutablePatternModifier;
 
 // Export an explicit template instantiation of the LocalPointer that is used as a
 // data member of ParameterizedModifier.
@@ -27,10 +22,16 @@ class MutablePatternModifier;
 #if defined (_MSC_VER)
 // Ignore warning 4661 as LocalPointerBase does not use operator== or operator!=
 #pragma warning(suppress: 4661)
-template class U_I18N_API LocalPointerBase<ParameterizedModifier>;
-template class U_I18N_API LocalPointer<ParameterizedModifier>;
+template class U_I18N_API LocalPointerBase<number::impl::ParameterizedModifier>;
+template class U_I18N_API LocalPointer<number::impl::ParameterizedModifier>;
 #endif
 
+namespace number {
+namespace impl {
+
+// Forward declaration
+class MutablePatternModifier;
+
 // Exported as U_I18N_API because it is needed for the unit test PatternModifierTest
 class U_I18N_API ImmutablePatternModifier : public MicroPropsGenerator, public UMemory {
   public:
