diff --git a/source/i18n/double-conversion-utils.h b/source/i18n/double-conversion-utils.h
index 02795b4b..4f48758e 100644
--- a/source/i18n/double-conversion-utils.h
+++ b/source/i18n/double-conversion-utils.h
@@ -97,6 +97,8 @@ inline void abort_noreturn() { abort(); }
 #else
 #undef DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS
 #endif  // _WIN32
+#elif U_PLATFORM == U_PF_BROWSER_NATIVE_CLIENT
+#undef DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS
 #else
 #error Target architecture was not detected as supported by Double-Conversion.
 #endif
