diff --git a/source/i18n/unicode/dtitvfmt.h b/source/i18n/unicode/dtitvfmt.h
index d4b9f936..42d77d04 100644
--- a/source/i18n/unicode/dtitvfmt.h
+++ b/source/i18n/unicode/dtitvfmt.h
@@ -60,7 +60,7 @@ class U_I18N_API FormattedDateInterval : public UMemory, public FormattedValue {
      * Default constructor; makes an empty FormattedDateInterval.
      * @draft ICU 64
      */
-    FormattedDateInterval() : fData(nullptr), fErrorCode(U_INVALID_STATE_ERROR) {};
+    FormattedDateInterval() : fData(nullptr), fErrorCode(U_INVALID_STATE_ERROR) {}
 
     /**
      * Move constructor: Leaves the source FormattedDateInterval in an undefined state.
diff --git a/source/i18n/unicode/formattedvalue.h b/source/i18n/unicode/formattedvalue.h
index ce59335e..60f97edd 100644
--- a/source/i18n/unicode/formattedvalue.h
+++ b/source/i18n/unicode/formattedvalue.h
@@ -122,7 +122,7 @@ class U_I18N_API ConstrainedFieldPosition : public UMemory {
      */
     inline int32_t getCategory() const {
         return fCategory;
-    };
+    }
 
     /**
      * Gets the field for the current position.
@@ -135,7 +135,7 @@ class U_I18N_API ConstrainedFieldPosition : public UMemory {
      */
     inline int32_t getField() const {
         return fField;
-    };
+    }
 
     /**
      * Gets the INCLUSIVE start index for the current position.
@@ -147,7 +147,7 @@ class U_I18N_API ConstrainedFieldPosition : public UMemory {
      */
     inline int32_t getStart() const {
         return fStart;
-    };
+    }
 
     /**
      * Gets the EXCLUSIVE end index stored for the current position.
@@ -159,7 +159,7 @@ class U_I18N_API ConstrainedFieldPosition : public UMemory {
      */
     inline int32_t getLimit() const {
         return fLimit;
-    };
+    }
 
     ////////////////////////////////////////////////////////////////////
     //// The following methods are for FormattedValue implementers; ////
diff --git a/source/i18n/unicode/listformatter.h b/source/i18n/unicode/listformatter.h
index 121dd192..9ce8ec86 100644
--- a/source/i18n/unicode/listformatter.h
+++ b/source/i18n/unicode/listformatter.h
@@ -85,7 +85,7 @@ class U_I18N_API FormattedList : public UMemory, public FormattedValue {
      * Default constructor; makes an empty FormattedList.
      * @draft ICU 64
      */
-    FormattedList() : fData(nullptr), fErrorCode(U_INVALID_STATE_ERROR) {};
+    FormattedList() : fData(nullptr), fErrorCode(U_INVALID_STATE_ERROR) {}
 
     /**
      * Move constructor: Leaves the source FormattedList in an undefined state.
diff --git a/source/i18n/unicode/reldatefmt.h b/source/i18n/unicode/reldatefmt.h
index 84d7c0a3..32344cd5 100644
--- a/source/i18n/unicode/reldatefmt.h
+++ b/source/i18n/unicode/reldatefmt.h
@@ -264,7 +264,7 @@ class U_I18N_API FormattedRelativeDateTime : public UMemory, public FormattedVal
      * Default constructor; makes an empty FormattedRelativeDateTime.
      * @draft ICU 64
      */
-    FormattedRelativeDateTime() : fData(nullptr), fErrorCode(U_INVALID_STATE_ERROR) {};
+    FormattedRelativeDateTime() : fData(nullptr), fErrorCode(U_INVALID_STATE_ERROR) {}
 
     /**
      * Move constructor: Leaves the source FormattedRelativeDateTime in an undefined state.
