diff --git a/source/common/locmap.cpp b/source/common/locmap.cpp
index a6311343..d075d031 100644
--- a/source/common/locmap.cpp
+++ b/source/common/locmap.cpp
@@ -32,10 +32,12 @@
 #include "cmemory.h"
 #include "unicode/uloc.h"
 
+#if 0
 #if U_PLATFORM_HAS_WIN32_API && UCONFIG_USE_WINDOWS_LCID_MAPPING_API
 #include <windows.h>
 #include <winnls.h> // LCIDToLocaleName and LocaleNameToLCID
 #endif
+#endif
 
 /*
  * Note:
